packages/neos-ui/src/manifest.js

```diff
@@ -428,4 +429,31 @@ manifest('main', {}, globalRegistry => {
                 break;
         }
     });
+
+    //
+    // When the server has removed a node, remove it as well from the store and the dom
+    //
+    serverFeedbackHandlers.add('Neos.Neos.Ui:RemoveNode', ({contextPath, parentContextPath}, store) => {
+        const state = store.getState();
+
+        if ($get('cr.nodes.focused.contextPath', state) === contextPath) {
+            store.dispatch(actions.CR.Nodes.unFocus());
+        }
+
+        if ($get('ui.pageTree.isFocused', state) === contextPath) {
+            store.dispatch(actions.UI.PageTree.focus(parentContextPath));
+        }
+
+        if ($get('ui.contentCanvas.contextPath', state) === contextPath) {
+            const parentNodeUri = $get(['cr', 'nodes', 'byContextPath', parentContextPath, 'uri'], state);
+
+            store.dispatch(actions.UI.ContentCanvas.setSrc(parentNodeUri));
+            store.dispatch(actions.UI.ContentCanvas.setContextPath(parentContextPath));
+        } else {
+            dom.findAll(`[data-__neos-node-contextpath="${contextPath}"]`)
+                .forEach(el => el.remove());
+        }
+
+        store.dispatch(actions.CR.Nodes.remove(contextPath));
```

Comment:
Yeah that immersive Dom mangling is nasty but hard to avoid. Something to keep in mind what one of the benefits of staying in declarative immutable paradise is :)